src/keyring/KeyringController.ts

```diff
@@ -561,9 +566,7 @@ export class KeyringController extends BaseController<
         async (keyring: KeyringObject, index: number): Promise<Keyring> => {
           const keyringAccounts = await keyring.getAccounts();
           const accounts = Array.isArray(keyringAccounts)
-            ? keyringAccounts.map((address) =>
-                ethUtil.toChecksumAddress(address),
-              )
+            ? keyringAccounts.map((address) => toChecksumAddress(address))
```

Comment:
is it possible that keyringAccounts could ever return non `0x` prefixed hex strings?

Reply:
I've added `toChecksumHexAddress` as discussed and used that here now.